File: apps/api/app/schemas/finance.py
from app.schemas.common import ORMModel

class TrustScoreRead(ORMModel):
    business_id: str
    score: int
    grade: str
    factors_json: dict
    proof_hash: str | None
    polygon_tx_hash: str | None

class FinanceReadinessRead(ORMModel):
    business_id: str
    verified_invoice_value: float
    completed_transactions: int
    dispute_rate: float
    recommended_limit: float
    report_json: dict

File: apps/web/lib/api/finance.ts
import { apiRequest } from './client';

export type TrustScore = {
  business_id: string;
  score: number;
  grade: string;
  factors_json: Record<string, unknown>;
};

export type FinanceReadiness = {
  business_id: string;
  verified_invoice_value: number;
  completed_transactions: number;
  dispute_rate: number;
  recommended_limit: number;
};

export const financeApi = {
  getScore: (businessId: string) => apiRequest<TrustScore>(`/finance/businesses/${businessId}/score`, { method: 'POST' }),
  getReadiness: (businessId: string) => apiRequest<FinanceReadiness>(`/finance/businesses/${businessId}/readiness`, { method: 'POST' }),
};
